mcodingbot/plugins/info.py

```diff
@@ -7,10 +7,22 @@
 plugin = Plugin()
 
 
+def natural_round(x: float) -> int:
+    """
+    Rounds in the same way as Python 2's `round()`.
+    If the real part of x is >= .5, rounds up ; else, rounds down.
+    """
+
+    n = int(x)
+    return n + (x - n >= 0.5)
```

Comment:
What about using a simple ternary?
```suggestion
    return x + 1 if x % 1 == .5 else x
```

```
nat_round_mod_ternary: executed 1000000 times in 0.14679307599999447 seconds.
```
# 

<br>

Here is an another way to write it, without the mod operator:
```suggestion
    return (y := int(x)) + (x - y >= .5)
```
```
nat_round_walrus_minus: executed 1000000 times in 0.14819169599991255 seconds.
```


other results (for reference):
```
round: executed 1000000 times in 0.07041415899993808 seconds.
int: executed 1000000 times in 0.04023135199986427 seconds.
nat_round: executed 1000000 times in 0.152997527000025 seconds.
```

Reply:
I love over-engineer features :D

Reply:
```suggestion
    return round(2 * x + .5) >> 1
```
is also a possibility. What do you think @Sigmanificient?

Reply:
> How is that last one working? xD

`round()` rounds 4.5 to 4 but 5.5 to 6.
If you multiply `x` by 2, you get rid of this even preference, and `+ .5` is to force rounding (it will disappear anyway since it rounds)
And the `>> 1` is just a fast `// 2` on integers

So for example:
`x = 4.5`
`round(9 + .5)` <=> `round(9.5)` == 10
10 >> 1 <=> 10 // 2 == 5, 4.5 rounded is 5

`x = 5.5`
`round(11 + .5)` <=> `round(11.5)` == 12
12 >> 1 == 6, 5.5 rounded is 6

Problem fixed :sunglasses: :+1:

Reply:
well i guess you won that one x)

Reply:
or you could even use `int(x+.5)`

Reply:
Sad that `&` and `|` dont work with float, could have been fun to mess with that

Reply:
> or you could even use `int(x+.5)`

doesn't work for negative numbers btw

Reply:
> > or you could even use `int(x+.5)`
> 
> doesn't work for negative numbers btw

nor my solution xd

Reply:
```suggestion
    if not n:
        return n
    return round(2 * n + n // abs(n) * .5) >> 1
```

Reply:
that last once seems slower with use of `//`? What does thez benchmark says about it?

Reply:
> that last one seems slower with use of `//`? What does the benchmark says about it?

way slower actually

Reply:
Well my evil one is quite slow too
```
nat_round_evil: executed 1000000 times in 0.2027178480002476 seconds.
```

Reply:
How about we just don't change it?